### ISSUE TITLE
Xonsh in Dolphin crashes with 'Application is already running'

### ISSUE THREAD

**Issue by jonaslb** [xonsh/xonsh#3529]
## Steps to Reproduce
Xonsh set as the default shell in Konsole. This should mean that Xonsh is also started when a terminal is started inside Dolphin (file manager) by pressing F4. This works the first time it's done. 
Try closing the in-file-manager terminal by pressing F4 again, and then re-open it by pressing F4 for the third time.

## Expected Behavior
Xonsh should resume.

## Current Behavior
An exception is raised, and it falls back to /bin/sh.

### Traceback 
I set `XONSH_DEBUG=1` in the shell/profile environment in Konsole, but it didn't seem to give more info.

<details>

```
$ Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/xonsh/main.py", line 405, in main
    return main_xonsh(args)
  File "/usr/lib/python3.7/site-packages/xonsh/main.py", line 434, in main_xonsh
    shell.shell.cmdloop()
  File "/usr/lib/python3.7/site-packages/xonsh/ptk2/shell.py", line 197, in cmdloop
    line = self.singleline(auto_suggest=auto_suggest)
  File "/usr/lib/python3.7/site-packages/xonsh/ptk2/shell.py", line 166, in singleline
    line = self.prompter.prompt(**prompt_args)
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/shortcuts/prompt.py", line 738, in prompt
    return run_sync()
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/shortcuts/prompt.py", line 727, in run_sync
    return self.app.run(inputhook=self.inputhook, pre_run=pre_run2)
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/application/application.py", line 709, in run
    return run()
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/application/application.py", line 681, in run
    f = self.run_async(pre_run=pre_run)
  File "/usr/lib/python3.7/site-packages/prompt_toolkit/application/application.py", line 527, in run_async
    assert not self._is_running, 'Application is already running.'
AssertionError: Application is already running.
Xonsh encountered an issue during launch                                                                                           
Failback to /bin/sh
sh-5.0$ sh-5.0$
```

</details>

## xonfig
<details>

```
$ xonfig                                                                                              
+------------------+---------------------+
| xonsh            | 0.9.17.dev2         |
| Git SHA          | 97a921c7            |
| Commit Date      | Apr 8 14:48:23 2020 |
| Python           | 3.7.6               |
| PLY              | 3.11                |
| have readline    | True                |
| prompt toolkit   | 2.0.9               |
| shell type       | prompt_toolkit2     |
| pygments         | 2.4.2               |
| on posix         | True                |
| on linux         | True                |
| distro           | fedora              |
| on darwin        | False               |
| on windows       | False               |
| on cygwin        | False               |
| on msys2         | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+
```

</details>

**Comment by scopatz** [comment]
Thanks for reporting.  This seems pretty strange

**Comment by jonaslb** [comment]
I imagine it could be related to #3628 -- I don't know if INT is being sent, but the traceback I get is identical to the one I get if I follow the steps in that issue.

**Comment by anki-code** [comment]
Closed as duplicate of #3628
